src/core/coap/secure_coap_server.cpp

```diff
@@ -115,6 +115,11 @@ void SecureServer::Receive(Message &aMessage, const Ip6::MessageInfo &aMessageIn
 
     if (!mNetif.GetDtls().IsStarted())
     {
+        Ip6::SockAddr sockAddr;
+        sockAddr.mAddress = aMessageInfo.GetPeerAddr();
+        sockAddr.mPort = aMessageInfo.GetPeerPort();
+        mSocket.Connect(sockAddr);
```

Comment:
Yes, I think so. #1520 deals with the source address of response packets. And this here prevents the secure server from receiving packets from another address during the DTLS session.